pkg/operator/controller/dns_status.go

```diff
@@ -34,8 +44,27 @@ func (r *reconciler) syncDNSStatus(dns *operatorv1.DNS, clusterIP, clusterDomain
 	return nil
 }
 
+// conditionsUpdatedWithinLast computes if the last update for progressing and degraded conditions
```

Comment:
Is it necessary to check the degraded status condition?  If a Corefile update causes enough pods to go unavailable at once that `computeDNSDegradedCondition` reports degraded, then it seems like that really is a problem.

Reply:
If there is a problem that lasts longer than 22s, then there will be reconciliations coming anyway as cluster keeps trying to fix the problem. In this case, we will set it to `Degraded=True` in one of the reconciliations after 22s. But, if a problem lasts less than 22s, then would it still make sense to set `Degraded=True` for a short time?